aws/aws-how-to/instances/upgrade-to-ubuntu-pro-at-scale-using-tokens-with-ssm.rst

```diff
@@ -0,0 +1,120 @@
+Upgrade to Ubuntu Pro at scale using tokens with SSM
+====================================================
+
+
+Overview
+---------
+
+Ubuntu Pro expands Ubuntu LTS’ 5 years of coverage up to ten-year of security coverage with an additional support of 23,000 packages beyond the main operating system and it is free for use on up to 5 machines!
+
+There are three ways to get Ubuntu Pro on AWS:
+
+    1. Launching an Ubuntu Pro machine from the AWS Marketplace, which is ready to use through a PAYG charging model.
+    2. Attaching an Ubuntu Pro token to any existing Ubuntu LTS server
+    3. Upgrading to Ubuntu Pro using AWS License Manager
+
+This how-to will cover the second case: Activating Pro via token using AWS Systems Manager.
+
+There are many ways to automate actions on AWS Systems Manager but for this how-to, we are going to use SSM Documents and the Run Command functionality.
+
+
+Prerequisites
+-------------
+
+- An AWS account
+- An Ubuntu One account (you can create one on `Ubuntu Pro | Ubuntu <https://ubuntu.com/pro>`_)
+- Machines with SSM agent installed (available by default on Ubuntu 16.04 onwards). If it is properly configured, the servers will appear as Managed Nodes on AWS SSM Fleet Manager (see the reference links for more information)
+- Basic understanding about writing and running SSM Documents (see the reference links for more information)
+
+
+Get a token
+------------
+
+To get an Ubuntu Pro token, you need to create an account in `Ubuntu Pro | Ubuntu <https://ubuntu.com/pro>`_ to see your subscriptions and get your tokens. Anyone can use Ubuntu Pro for free on up to 5 machines, even for production workloads.
+
+Log-in by clicking on *Your subscriptions*. You will be redirected to the following screen where the token is exposed as shown in the orange ellipse:
+
+.. image:: upgrade-to-ubuntu-pro-at-scale-using-tokens-with-ssm-images/0_token_screen.png
+   :align: center
+
+
+Create a SSM Document to attach the token
+------------------------------------------
+
+The SSM Document is just a JSON or YAML file containing the script in steps or sections.
+
+    Note:
+    As the token is personal, it is not recommended to hard-code it into the code but add it as a user parameter.
+
+This is the full code:
+
+.. code-block::
+
+    ---
+    schemaVersion: '2.2'
+    description: "Enable Pro services using a Pro token"
+    parameters:
+    token:
+        type: String
+    mainSteps:
+    - action: aws:runShellScript
+    name: update
+    inputs:
+        runCommand:
+        - "sudo apt-get update && sudo apt-get upgrade -y"
+    - action: aws:runShellScript
+    name: attachToken
+    inputs:
+        runCommand:
+        - "sudo pro attach {{ token }}"
+    - action: aws:runShellScript
+    name: updateWithESM
+    inputs:
+        runCommand:
+        - "sudo apt-get update && sudo apt-get upgrade -y"
+
+
+The token will be requested to the user when running this document, then it will execute the rest of the code:
```

Comment:
This sentence needs re-wording. "The token will be requested to the user" is unclear. Also unclear is what is meant by "it" in "then it will execute the rest of the code."